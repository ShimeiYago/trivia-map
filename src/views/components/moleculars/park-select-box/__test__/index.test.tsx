import { shallow, ShallowWrapper } from 'enzyme';
import { ParkSelectBox, Props } from '..';

let wrapper: ShallowWrapper<Props, null, ParkSelectBox>;

const defaultProps: Props = {
  park: 'L',
  onChangePark: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<ParkSelectBox {...defaultProps} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sea', () => {
    wrapper.setProps({
      park: 'S',
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('handleChange', () => {
  it('call onChangePark with "L"', () => {
    const instance = wrapper.instance();

    const event = {
      target: {
        checked: false,
      },
    };

    instance['handleChange'](event as React.ChangeEvent<HTMLInputElement>);
    expect(instance.props.onChangePark).toBeCalledWith('L');
  });

  it('call onChangePark with "S"', () => {
    const instance = wrapper.instance();

    const event = {
      target: {
        checked: true,
      },
    };

    instance['handleChange'](event as React.ChangeEvent<HTMLInputElement>);
    expect(instance.props.onChangePark).toBeCalledWith('S');
  });
});
