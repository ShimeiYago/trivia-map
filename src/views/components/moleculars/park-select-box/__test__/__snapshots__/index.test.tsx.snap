// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Paper)
  elevation={2}
  square={true}
>
  <ForwardRef(Stack)
    py={3}
    spacing={3}
  >
    <ForwardRef(Typography)
      align="center"
      component="div"
    >
      <Image
        src="logo-blue.png"
        width="100px"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Stack)
      component="nav"
      direction="row"
      divider={
        <ForwardRef(Divider)
          flexItem={true}
          orientation="vertical"
        />
      }
      justifyContent="center"
      spacing={2}
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      <Link
        key="footer-navi-マップ"
        to="/"
      >
        マップ
      </Link>
      <Link
        key="footer-navi-投稿一覧＆検索"
        to="/articles"
      >
        投稿一覧＆検索
      </Link>
      <Link
        key="footer-navi-お問い合わせ"
        to="/inquiry"
      >
        お問い合わせ
      </Link>
      <Link
        key="footer-navi-利用規約"
        to="/terms-of-service"
      >
        利用規約
      </Link>
      <Link
        key="footer-navi-プライバシーポリシー"
        to="/privacy-policy"
      >
        プライバシーポリシー
      </Link>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</ForwardRef(Paper)>
`;

exports[`Shallow Snapshot Tests mobile 1`] = `
<ForwardRef(Paper)
  elevation={2}
  square={true}
>
  <ForwardRef(Stack)
    py={3}
    spacing={3}
  >
    <ForwardRef(Typography)
      align="center"
      component="div"
    >
      <Image
        src="logo-blue.png"
        width="100px"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Stack)
      component="nav"
      direction="row"
      divider={
        <ForwardRef(Divider)
          flexItem={true}
          orientation="vertical"
        />
      }
      justifyContent="center"
      spacing={2}
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      <Link
        key="footer-navi-マップ"
        to="/"
      >
        マップ
      </Link>
      <Link
        key="footer-navi-投稿一覧＆検索"
        to="/articles"
      >
        投稿一覧＆検索
      </Link>
      <Link
        key="footer-navi-お問い合わせ"
        to="/inquiry"
      >
        お問い合わせ
      </Link>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      component="nav"
      direction="row"
      divider={
        <ForwardRef(Divider)
          flexItem={true}
          orientation="vertical"
        />
      }
      justifyContent="center"
      spacing={2}
      sx={
        Object {
          "fontSize": 14,
        }
      }
    >
      <Link
        key="footer-navi-利用規約"
        to="/terms-of-service"
      >
        利用規約
      </Link>
      <Link
        key="footer-navi-プライバシーポリシー"
        to="/privacy-policy"
      >
        プライバシーポリシー
      </Link>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</ForwardRef(Paper)>
`;
